Source/WebKit/UIProcess/Extensions/WebExtension.cpp

```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+                if (matchPattern->isSupported())
+                    excludeMatchPatterns.add(matchPattern.releaseNonNull());
+            }
+        }
+
+        // Optional. Applied after matches to include only those URLs that also match this glob.
+        auto includeGlobPatternStrings = injectedContentObject->getArray(contentScriptsIncludeGlobsManifestKey);
+        if (!includeGlobPatternStrings)
+            includeGlobPatternStrings = JSON::Array::create();
+
+        includeGlobPatternStrings = filterObjects(*includeGlobPatternStrings, [](auto& value) {
+            return !!value.asString().isEmpty();
+        });
+
+        // Optional. Applied after matches to exclude URLs that match this glob.
+        auto excludeGlobPatternStrings = injectedContentObject->getArray(contentScriptsExcludeGlobsManifestKey);
+        if (!excludeGlobPatternStrings)
+            excludeGlobPatternStrings = JSON::Array::create();
+
+        excludeGlobPatternStrings = filterObjects(*excludeGlobPatternStrings, [](auto& value) -> bool {
+            return !!value.asString().isEmpty();
+        });
+
+        // Optional. The "all_frames" field allows the extension to specify if JavaScript and CSS files should be injected into all frames matching the specified URL requirements or only into the
+        // topmost frame in a tab. Defaults to false, meaning that only the top frame is matched. If specified true, it will inject into all frames, even if the frame is not the topmost frame in
+        // the tab. Each frame is checked independently for URL requirements, it will not inject into child frames if the URL requirements are not met.
+        auto injectsIntoAllFramesBool = injectedContentObject->getBoolean(contentScriptsAllFramesManifestKey);
+        bool injectsIntoAllFrames = injectsIntoAllFramesBool ? *injectsIntoAllFramesBool : false;
+
+        InjectionTime injectionTime = InjectionTime::DocumentIdle;
+        String runsAtString = injectedContentObject->getString(contentScriptsRunAtManifestKey);
+        if (!runsAtString || (runsAtString == contentScriptsDocumentIdleManifestKey))
+            injectionTime = InjectionTime::DocumentIdle;
+        else if (runsAtString == contentScriptsDocumentStartManifestKey)
+            injectionTime = InjectionTime::DocumentStart;
+        else if (runsAtString == contentScriptsDocumentEndManifestKey)
+            injectionTime = InjectionTime::DocumentEnd;
+        else
+            recordError(createError(Error::InvalidContentScripts, WEB_UI_STRING("Manifest `content_scripts` entry has unknown `run_at` value.", "WKWebExtensionErrorInvalidContentScripts description for unknown 'run_at' value")));
+
+        WebExtensionContentWorldType contentWorldType = WebExtensionContentWorldType::ContentScript;
+        String worldString = injectedContentObject->getString(contentScriptsWorldManifestKey);
+        if (!worldString || runsAtString == contentScriptsIsolatedManifestKey)
```

Comment:
This one is important (was comparing `runsAtString` instead of `worldString`):
```suggestion
        auto contentWorldType = WebExtensionContentWorldType::ContentScript;
        auto worldString = injectedContentObject->getString(contentScriptsWorldManifestKey);
        if (!worldString || worldString == contentScriptsIsolatedManifestKey)
```